src/lib/toolbar/toolbar.ts

```diff
@@ -44,10 +39,35 @@ export const _MdToolbarMixinBase = mixinColor(MdToolbarBase);
   changeDetection: ChangeDetectionStrategy.OnPush,
   encapsulation: ViewEncapsulation.None
 })
-export class MdToolbar extends _MdToolbarMixinBase implements CanColor {
+export class MdToolbar extends _MdToolbarMixinBase implements CanColor, AfterViewInit {
 
-  constructor(renderer: Renderer2, elementRef: ElementRef) {
+  /** Element reference to the default first row of the toolbar. */
+  @ViewChild('defaultFirstRow', { read: ElementRef }) _defaultFirstRow: ElementRef;
+
+  constructor(renderer: Renderer2, elementRef: ElementRef, private _platform: Platform) {
     super(renderer, elementRef);
   }
 
+  ngAfterViewInit() {
+    // Do nothing if we're not on the browser platform.
+    if (!this._platform.isBrowser) {
+      return;
+    }
+
+    const firstRowElement = this._defaultFirstRow.nativeElement;
+
+    // If the first toolbar row, which will be auto-generated, does not have any projected content,
+    // then the auto-generated first toolbar row should be removed. This gives developers
+    // full control over the first toolbar row (e.g. for use with flex-layout).
+    if (!firstRowElement.innerHTML.trim()) {
```

Comment:
Instead of `innerHTML`, check `!firstRow.textContent.trim() && !firstRow.childNodes.length` ?
(I don't even like looking at `innerHTML` if it can be avoided)

Reply:
Yeah pretty sure that works too. Let's see if we actually still need that check with the discussed changes on Slack.